src/libexpr/primops/fetchTree.cc

```diff
@@ -48,10 +51,40 @@ void emitTreeAttrs(
     v.attrs->sort();
 }
 
-static void prim_fetchTree(EvalState & state, const Pos & pos, Value * * args, Value & v)
+std::string fixURI(std::string uri, EvalState &state)
+{
+    state.checkURI(uri);
+    return uri.find("://") != std::string::npos ? uri : "file://" + uri;
+}
+
+void addURI(EvalState &state, fetchers::Attrs &attrs, Symbol name, std::string v)
+{
+    string n(name);
+    attrs.emplace(name, n == "url" ? fixURI(v, state) : v);
+}
+
+void validateFetcherAttrs(fetchers::Attrs attrs, const Pos &pos)
 {
-    settings.requireExperimentalFeature("flakes");
```

Comment:
We need to keep this in `fetchTree`.

Reply:
Why though? At least `fetchGit` works (until now) without flakes.

Reply:
Because `fetchTree` (not `fetchGit`) needs to be behind an experimental feature flag.

Reply:
Oh, you're absolutely right, fixed. Sorry.